components/camel-weka/src/main/java/org/apache/camel/component/weka/WekaEndpoint.java

```diff
@@ -64,11 +92,245 @@ public boolean isSingleton() {
+        
+        Dataset dataset = assertDatasetBody(exchange);
+        String fpath = configuration.getPath();
+        
+        if (fpath != null) {
+            
+            dataset.write(Paths.get(fpath));
+            return dataset;
+            
+        } else {
+            
+            // The internal implementation of DataSink does this.. 
+            // Instances.toString().getBytes()
+            //
+            // Therefore, we avoid creating yet another copy of the
+            // instance data and call Instances.toString() as well
+            
+            Instances instances = dataset.getInstances();
+            byte[] bytes = instances.toString().getBytes();
+            return new ByteArrayInputStream(bytes);
+        }
+    }
+
+    Dataset handleFilterCmd(Exchange exchange) throws Exception {
+        
+        String applyValue = configuration.getApply();
+
+        Dataset dataset = assertDatasetBody(exchange);
+        dataset = dataset.apply(applyValue);
+        
+        return dataset;
+    }
+
+    Dataset handleModelCmd(Exchange exchange) throws Exception {
+        
+        Dataset dataset = assertDatasetBody(exchange);
+        
+        String dsname = configuration.getDsname();
+        boolean crossValidate = configuration.isXval();
+        String buildSpec = configuration.getBuild();
+        String loadFrom = configuration.getLoadFrom();
+        String saveTo = configuration.getSaveTo();
+        
+        // Load the Model
+        
+        if (loadFrom != null) {
+            
+            Classifier cl = dataset
+                    .loadClassifier(new ModelLoader(loadFrom))
+                    .getClassifier();
+            
+            AssertState.notNull(cl, "Cannot load the classifier from: " + loadFrom);
+            LOG.info("{}", cl);
+        } 
+        
+        // Build a classifier
+        
+        else if (buildSpec != null) {
+            
+            dataset.buildClassifier(buildSpec);
+            
+            // Cross Validate the Model
+            
+            if (crossValidate) {
+                int seed = configuration.getSeed();
+                int folds = configuration.getFolds();
+                dataset.crossValidateModel(folds, seed);
+            } 
+            
+            // Validate the Model using explicit/current instances
+            
+            else {
+                
+                // Use the named data set training
+                if (dsname != null) {
+                    dataset.pop(dsname);
+                }
+                
+                // Train with current instances
+                dataset.evaluateModel();
+            }
+            
+            Classifier cl = dataset.getClassifier();
+            AssertState.notNull(cl, "Model command requires 'load' or 'apply'");
+            LOG.info("{}", cl);
+            
+            Evaluation ev = dataset.getEvaluation();
+            LOG.info("{}", ev.toSummaryString());
```

Comment:
Is this needed to be logged at info level? And is this per exchange which then can get noisy

Reply:
Done